app/models/bot/tagger.rb

```diff
@@ -0,0 +1,65 @@
+require 'json'
+
+class Bot::Tagger < BotUser
+  check_settings
+  class Error < ::StandardError
+  end
+
+  def self.get_tag_text(tag_id,auto_tag_prefix,ignore_autotags)
+    tag=TagText.find_by_id(tag_id).text
+    if tag.nil? || (ignore_autotags && tag[0]==auto_tag_prefix)
+      return nil
+    else
+      tag[0]==auto_tag_prefix ? tag[1..] : tag
+    end
+  end
+
+  def self.run(body)
+    Rails.logger.info("[AutoTagger Bot] Received event with body of #{body}")
+    if CheckConfig.get('alegre_host').blank?
+      Rails.logger.warn("[AutoTagger Bot] Skipping events because `alegre_host` config is blank")
+      return false
+    end
+
+    handled = false
+    pm = nil
+    begin
+      settings=JSON.parse(body["settings"])
+      auto_tag_prefix=settings["auto_tag_prefix"]
+      threshold=settings["threshold"]/100.0
+      ignore_autotags=settings["ignore_autotags"]
+      pm = ProjectMedia.where(id: body.dig(:data, :dbid)).last
+      if body.dig(:event) == 'create_project_media' && !pm.nil?
+        Rails.logger.info("[AutoTagger Bot] [ProjectMedia ##{pm.id}] This item was just created, processing...")
+        results=Bot::Alegre.get_items_with_similar_text(pm, Bot::Alegre::ALL_TEXT_SIMILARITY_FIELDS,
```

Comment:
**issue** this line has a lot of complexity - can it be broken down? a comment would also be helpful, I think

Reply:
Comments added.